rfcs/react-components/components/Spinner.md

```diff
@@ -0,0 +1,147 @@
+| Polaris                 | `Spinner`          | [Spinner](https://polaris.shopify.com/components/feedback-indicators/spinner)             | Default Spinner with a size prop, with preset small and large values                                                                                                                                                                                                                              |
+| Lightning Design System | `Spinner`          | [Spinner](https://react.lightningdesignsystem.com/components/spinners/)                   | A more robust indeterminate Spinner with props that allow the Spinner to be rendered in a container, add styling to said container, inline the Spinner, and change the size                                                                                                                       |
+
+### Comparison of v8 and v0
+
+The existing components are:
+
+- v8
+  - `Spinner`
+- v0
+  - `Loader`
+- Both `Spinner` and `Loader` have very similar props
+- `Loader` has `delay`, while `Spinner` has aria props exposed for accessibility.
+
+## Sample Code
+
+Basic examples:
+
+```jsx=
+import { Spinner } from "@fluentui/react-spinner";
+
+function App() {
+    return (
+        <Spinner size="medium" label="Loading" />
+    )
+}
+
+```
+
+## Variants
+
+- Determinate Spinner
+  - Add a `value` prop that prefills the `Spinner` to a specific percentage. Can be implemented using a value from 0 to 1.
+- Indeterminate Spinner
+  - The default Spinner that spins around indefinitely
+
+### Shape
+
+The Spinner is represented as a circle with an arc of a darker shade rotating the circumfrence
+
+## API
+
+From [Spinner.types.tsx]()
+
+### Slots
+
+- `root` - The root element of the Spinner. The html element is a `span`
+- `loader` - The svg element that gets animated into a Spinner. The html element is `svg`
+- `label` - The text shown with the Spinner. This uses the `Text` control
+
+### Props
+
+```jsx
+export type SpinnerProps = ComponentProps & React.HTMLAttributes<HTMLElement> & {
+  /* The appearance of the Spinner*/
+  appearance?: "primary" | "inverted";
+  /* ariaLabel prop for accessibility */
+  ariaLabel?: string;
+  /* The label prop allows user to add text. Defaults to "Loading..." */
+  label?: <Text />;
+  /* The labelPosition prop allows user to set the location of the label*/
+  labelPosition?: "above" | "below" | "before" | "after";
+  // new prop for vertical? : boolean. Will allow for the labelPosition prop to only need to be "before" and "after"
+  vertical?: boolean;
+  /* The size prop sets the size of the Spinner
+   * @defaultValue "medium"*/
+  size?: "tiny" | "x-small" | "small" | "medium" | "large" | "x-large" | "huge";
+  // inactive ? : boolean
+  status?: "active" | "success" | "error" | "inactive";
+  /** The indeterminate prop sets whether the Spinner is a determinate Spinner or indetermindate Spinner*/
```

Comment:
Can you add a defaultValue for `indeterminate`?

Reply:
What do you think about dropping the `indeterminate` property? I'd assume that, with proper docs, any user setting the `value` prop expects to have to manage the value themselves.
`value` has no purpose if `indeterminate` is true, right?

Reply:
If we keep this prop or similar, the description should include something about what determinate/indeterminate mean in this context. :)

Also I'm guessing @spmonahan's comment was referring to a `@defaultValue` tag not a `defaultValue` prop in this context, since a spinner doesn't really manage a value?

Reply:
You're right @ecraig12345 . Sorry I wasn't clear there! Though after reading @andrefcdias 's comment I'm glad I wasn't so clear because it's made me think of another thing :)

If `Spinner` has a boolean `indeterminate` prop that implies when it is in a determinate state it must have a value (unless I'm misunderstanding what `indeterminate` means in this context?) and this spec doesn't currently have any props for managing a value. With the current `Spinner` spec I think this can be accomplished using [`aria-valuenow/min/max`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role) but given that these props don't appear directly in the type this feels "hidden" from users. If `Spinner` will support a determinate state it seems we need to consider how to provide values to it. Maybe something like the [`max`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress#attr-max) and [`value`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress#attr-value) attributes on the native `progress` element.

Or maybe a better place to start is to remove the `indeterminate` prop as @andrefcdias suggested and have `Spinner` always be indeterminate?

Reply:
I agree that we should remove `indeterminate` if the user provides a `value` prop then it should always be determinate.

I would also suggest that you add `value` to the props types in the spec since you mention it in  the variant section already

Reply:
I think it makes sense to make `Spinner` only be indeterminate, as was the case in v8.